### PR TITLE
revert subtitution of "---" for "None"

### PR DESCRIPTION
**Pull request by Be-ing**
Reverting the functional changes of #3485 and keeping the refactoring

### CODE REVIEW

**Comment by ronso0** [comment]
my motivation to have a clearly distinguishable state. Shapes are much quicker to recognise than text.
None may be at the end or top of the affected lists which may make it easy to find but for closed comboboxes this is not relevant.
--- is a very simple shape and you don't have to read None to know an effect slot is empty.
In the collapsed effect units it's so much clearer now. 
A compromise would be to have None in the expanded list, and --- when they're closed. But the effort of refactoring the effect selector or any Pref combobox was worth it. 

I'd agree if you revert the change for Pref > Effects an EQ but please just leave it as it in the skin widgets. 

Same for the Sound API:
--- vs None in a list that also has similiar short items like Alsa
Same for the item in the sound in/output configuration below
--- vs None so that it's much easier to spot if all devices are set, respectively that unwanted in/outputs are not accidentally set.

Btw the english None is shorter than most translations that come to mind.

**Comment by Be-ing** [comment]
IMO it is more important that the meaning is clear than it looks visually distinct. Why does it need to look so distinct? Why is it important that an empty effect slot stands out from other others? To me that is a little bit distracting.

The problem with the Sound Hardware preferences is that there are so many comboboxes shown at once. The solution to that problem is a redesign

**Comment by ronso0** [comment]
> IMO it is more important that the meaning is clear than it looks visually distinct. Why does it need to look so distinct? Why is it important that an empty effect slot stands out from other others?

so that the effect selectors fulfill their purpose: display the current status of loaded effects. 
the reason for me using --- was that I engaged effects without noticing at a glance they were empty (due to unnoticed Shift+Meta knob turn earlier, another UX issue IMO, disabled now).
With --- that's unlikely to happen again.

**Comment by ronso0** [comment]
> IMO it is more important that the meaning is clear than it looks visually distinct. Why does it need to look so distinct? Why is it important that an empty effect slot stands out from other others? To me that is a little bit distracting.
> 
> The problem with the Sound Hardware preferences is that there are so many comboboxes shown at once. The solution to that problem is a redesign

yes, we're touching that topic over and over again without someone starting a real design discussion (in the right moment).
this was a small step to improve UX. 

As I said,  you feel free to change it back to None in the Effects and EQ preferences if it really bothers you.

**Comment by Be-ing** [comment]
>  the reason for me using --- was that I engaged effects without noticing at a glance they were empty (due to unnoticed Shift+Meta knob turn earlier, another UX issue IMO, disabled now).

Okay, thank you for explaining.

> As I said, you feel free to change it back to None in the Effects and EQ preferences if it really bothers you.

Will do.

**Comment by Holzhaus** [comment]
How about using empty string instead of `---`? It's both visually distinct and the meaning is clear.

**Comment by daschuer** [comment]
You can't really good select something empty in a combo box. If you have the dashes, your eyes have something to hook on. 
I think the dashes are some kind of industrial standard for "no value". The value is "dashed out". 
At least in the embedded display context, dashed lines are used for indicating that the display is OK, but that there is just no value.

> I don't like this change. This feels like a regression. What was ever the problem with "None"? What was the motivation for changing this? Have you done any usability tests comparing this with "None"? I cannot recall any problems ever caused by the string "None". Who would know what --- means in this context? This looks bizarre IMO.

An empty string is even worse in this context.

**Comment by Be-ing** [comment]
An empty string looks odd in a combobox. It seems like it could be an error instead of a valid value.

**Comment by uklotzde** [comment]
I conside `---` a recognizable placeholder. An alternative could be `<None>`. Angle brackets are often used for such placeholders and special values, e.g. `<Keep>`, `<Clear>`, or `<Remove>` when batch editing fields with different values from multiple entities.

**Comment by Be-ing** [comment]
Angle brackets look like XML.

**Comment by Be-ing** [comment]
Anyway, can we all agree to go back to "None" for the EQ preferences, merge this, and move on?

**Comment by ronso0** [comment]
> Anyway, can we all agree to go back to "None" for the EQ preferences, merge this, and move on?

yup.
any objections?

File: src/preferences/dialog/dlgprefeq.cpp [hunk 1]
```diff
@@ -23,6 +23,8 @@ const QString kSingleEq = "SingleEQEffect";
 const QString kDefaultEqId = BiquadFullKillEQEffect::getId();
 const QString kDefaultMasterEqId = QString();
 const QString kDefaultQuickEffectId = FilterEffect::getId();
+//: Shown when no EQ is loaded
+const QString kNoEffectLoadedLabel = QObject::tr("None");
```
Comment (ronso0) [inline src/preferences/dialog/dlgprefeq.cpp:27]: recently I used  [Disambiguation string](https://doc.qt.io/qt-5/i18n-source-translation.html#disambiguation) to add a hint for translators. Then I discovered `//:` is similiar.
What's the difference for transifex translators?

**Comment by Holzhaus** [comment]
> > Anyway, can we all agree to go back to "None" for the EQ preferences, merge this, and move on?
> 
> yup.
> any objections?

Yes. I think this inconsistency does not make sense.

IIRC it's possible to use rich-text in these comboboxes. Can we maybe make "None" in italics or something, and then use that everywhere? And maybe use braces, e.g. *(None)* ?

**Comment by Be-ing** [comment]
The EQ preferences are in a different context from the skin widgets. I don't think there's a problem not having them consistent.

**Comment by Holzhaus** [comment]
The EQ preferences should at least be consistent with all other preferences comboboxes, e.g. the sound preferences.

**Comment by ronso0** [comment]
> Yes. I think this inconsistency does not make sense.

okay, so you'd like to have all preferecnes comboboxes consistent? or is it about skin effect widgets vs. Effect/EQ comboboxes?

As explained above I find --- much better in the Sound preferences where one needs to grasp the overall config at a glance.
_(None)_ wouldn't help much IMO. For my previous PR I really couldn't come up with any other _graphical_ placeholder than ---

Would it help to have all comboboxes consistent by adding the empty item at the **top**? Currently the position differs.
Sound API > top
Sound devices > top
EQ effects > bottom
effect selector widget > bottom (list may be long and bottom items may be hidden)

@Be-ing before you explain that in capitals, I'm aware such a refactoring was not your intention here ;) I could take over if you like..

**Comment by Be-ing** [comment]
I don't care enough about this to continue discussing it.

**Pull request closed without merging**